File: lab1/sesion-2/ejemplos_c/conceptos_c/estructuras/comparacion_lenguajes/java/Ejemplo5.java
public class Ejemplo5 {

    // Método que recibe una referencia a un objeto Persona
    public static void imprimir(Persona p) {
        // Se accede a los atributos con el operador punto "."
        System.out.printf("%s tiene %d años\n", p.nombre, p.edad);
    }

    public static void main(String[] args) {
        Persona p1 = new Persona();
        p1.nombre = "Carlos";
        p1.edad = 28;

        // Se pasa la referencia del objeto al método
        imprimir(p1);
    }
}

File: lab1/sesion-2/ejemplos_c/conceptos_c/estructuras/comparacion_lenguajes/python/ejemplo5.py
class Persona:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

# La función recibe una referencia a un objeto Persona
def imprimir(p):
    # Se accede a los atributos con el operador punto "."
    print(f"{p.nombre} tiene {p.edad} años")


if __name__ == "__main__":
    p1 = Persona("Carlos", 28)

    # Se pasa la referencia del objeto a la función
    imprimir(p1)